core/src/serve_repair.rs

```diff
@@ -269,19 +269,23 @@ impl ServeRepair {
         Ok(out)
     }
 
-    pub fn repair_request(&self, repair_request: &RepairType) -> Result<(SocketAddr, Vec<u8>)> {
+    pub fn repair_request(
+        &self,
+        cluster_slots: &ClusterSlots,
+        repair_request: &RepairType,
+    ) -> Result<(SocketAddr, Vec<u8>)> {
         // find a peer that appears to be accepting replication and has the desired slot, as indicated
         // by a valid tvu port location
-        let valid: Vec<_> = self
+        let repair_peers: Vec<_> = self
             .cluster_info
             .read()
             .unwrap()
             .repair_peers(repair_request.slot());
```

Comment:
hmm, it might be time we start caching this somewhere, we do this `repair_peers` calculation on every single individual repair request :(